src/sqs/sqs.ts

```diff
@@ -267,8 +271,12 @@ export class SqsQueue<MessagePayload> {
 		try {
 			const messageProcessingStartTime = Date.now();
 
-			if (await preemptiveRateLimitCheck(context, this)) {
-				context.log.info("Preemptive rate limit threshold exceeded");
+			const rateLimitCheckResult = await preemptiveRateLimitCheck(context, this);
+			if (rateLimitCheckResult.isExceedThreshold) {
+				//threshold exceeded, retry via sending a new message.
+				const { MessageId } = await this.sendMessage(payload, rateLimitCheckResult.resetTimeInSeconds, context.log);
+				await this.deleteMessage(context);
+				context.log.info({ NewMessageId: MessageId, DeletedMessageId: message.MessageId }, "Preemptive rate limit threshold exceeded, rescheduled new one and deleted the origin msg");
```

Comment:
nit: same, I think we have a different naming convention for the object properties

Reply:
fixed.